csharp/extractor/Semmle.Extraction.CSharp/Entities/Types/TypeParameterConstraints.cs

```diff
@@ -1,14 +1,65 @@
+using Microsoft.CodeAnalysis;
 using System.IO;
 
 namespace Semmle.Extraction.CSharp.Entities
 {
-    internal class TypeParameterConstraints : FreshEntity
+    internal class TypeParameterConstraints : CachedEntity<ITypeParameterSymbol>
     {
-        public TypeParameterConstraints(Context cx)
-            : base(cx) { }
+        private readonly TypeParameter parent;
 
-        protected override void Populate(TextWriter trapFile)
+        public TypeParameterConstraints(Context cx, TypeParameter parent)
+            : base(cx, parent.Symbol)
         {
+            this.parent = parent;
+        }
+
+        public override void WriteId(EscapingTextWriter trapFile)
+        {
+            trapFile.WriteSubId(parent);
+            trapFile.Write(";typeparameterconstraints");
```

Comment:
Could you remind me, why this is needed (we are only doing this for entities and not fresh entities).

Reply:
It is to give a unique ID to the entity. If we simply did `trapFile.WriteSubId(parent);` it would overlap with the ID of the parent.

Reply:
Ok, so the ID we write into the trap, is basically the same as the parent id except that we postfix it with ";typeparameterconstraints"?

Reply:
Yes, exactly.